CodeGeneration/Tests/ValidateSyntaxNodes/ValidateSyntaxNodes.swift

```diff
@@ -178,16 +178,16 @@ class ValidateSyntaxNodes: XCTestCase {
         }
       }
 
-    case .token(tokenKind: "IdentifierToken"), .token(tokenKind: "IntegerLiteralToken"), .token(tokenKind: "FloatLiteralToken"):
+    case .token(tokenKind: .identifier), .token(tokenKind: .integerLiteral), .token(tokenKind: .floatLiteral):
       // We allow arbitrary naming of identifiers and literals
       break
-    case .token(tokenKind: "CommaToken"):
+    case .token(tokenKind: .comma):
       if child.name != "TrailingComma" && child.name != "Comma" {
         return "child '\(child.name)' has a comma keyword as its only token choice and should thus be named 'Comma' or 'TrailingComma'"
       }
     case .token(tokenKind: let tokenKind):
       let expectedChildName =
-        tokenKind
+        tokenKind.tokenSpec.varOrCaseName.text.withFirstCharacterUppercased
```

Comment:
You can remove the `withFirstCharacterUppercased` here and just lowercase the child names in the expected failures of the `assertFailuresMatchXFails` calls. That cleans them up as well.

Reply:
I'm not entirely sure if I completely understand you, but I believe that your suggestion may not work. Consider an example like `binaryOperator`: we drop the `binary` prefix and expect `Operator`. Until we can lowercase the first letters of children's names, the current solution appears to be the most reliable

Reply:
Oh, sorry, you’re right. Also, I thought that we had already lowercased the child names, which isn’t true. So: Nothing to be done here.

Reply:
Speaking about lowercasing the child names check out DM's on swift forums... XD That's a pretty funny situation... XD